src/Nethermind/Nethermind.Consensus.Ethash/EthashSealValidator.cs

```diff
@@ -86,32 +86,47 @@ public void HintValidationRange(Guid guid, long start, long end)
             _ethash.HintRange(guid, start, end);
         }
 
-        public bool ValidateParams(BlockHeader parent, BlockHeader header)
+        public bool ValidateParams(BlockHeader parent, BlockHeader header, bool isUncle = false)
         {
-            bool extraDataNotTooLong = header.ExtraData.Length <= 32;
-            if (!extraDataNotTooLong)
+            return ValidateExtraData()
+                   && ValidateDifficulty()
+                   && (isUncle || ValidateTimestamp());
```

Comment:
Not sure if it matters for performance that those methods are closures or not, would have inspect generated code (like in sharplab) or just make then non-closures.